lambda/src/main/scala/pricemigrationengine/handlers/NotificationEmailHandler.scala

```diff
@@ -70,13 +72,30 @@ object NotificationEmailHandler {
           contact.IdentityID__c
         )
       )
+
+      _ <- updateCohortItemStatus(cohortItem.subscriptionName, EmailSendComplete)
     } yield ()
   }
 
   def requiredField(field: Option[String], fieldName: String): ZIO[Any, NotificationEmailHandlerFailure, String] = {
     ZIO.fromOption(field).orElseFail(NotificationEmailHandlerFailure(s"$fieldName is a required field"))
   }
 
+  def updateCohortItemStatus(subscriptionNumber: String, processingStage: CohortTableFilter) = {
+    for {
+      now <- clock.currentDateTime.mapError(ex => NotificationEmailHandlerFailure(s"Failed to get time: $ex"))
```

Comment:
Is it possible not to get current time?

Reply:
Are you thinking of passing it through from the top level?

I guess it wouldnt be quite as accurate, the email job may take some time to run, so the time stamp could be minute out of sync with when the email is actually sent.

I guess its not a big deal but whats the object to getting the time here?

Reply:
I was just commenting on the exception mapping.  I wouldn't think `clock.currentDateTime` could fail.

Reply:
Ah, well it is a side-effect, you never know :)

The clock is the Zio clock as such you have to handle the failure case. 

You are right it might be a bit unnecessary but Im using it to stub out the clocks in the test and Ive used it fairly consistently in the code.

Ill leave it as is, maybe we can go through and change everything over to use java.time.Clock instead.